api/src/data_migration/transformation/subtask/transform_competition.py

```diff
@@ -41,16 +42,25 @@ def transform_records(self) -> None:
 
                 if opportunity_cfda:
                     opportunity_id = opportunity_cfda.opportunity_id
-                    opportunity_assistance_listing_id = opportunity_cfda.opp_cfda_id
 
                     # Make sure the opportunity exists in our target table
                     opportunity = self.db_session.execute(
-                        select(Opportunity).where(Opportunity.opportunity_id == opportunity_id)
+                        select(Opportunity)
+                        .where(Opportunity.opportunity_id == opportunity_cfda.opportunity_id)
+                        .options(selectinload(Opportunity.opportunity_assistance_listings))
                     ).scalar_one_or_none()
 
-                    if not opportunity:
-                        opportunity_id = None
-                        opportunity_assistance_listing_id = None
+                    if opportunity:
```

Comment:
Could we add a comment above this to explain why we're doing this? Something like:

```
In order to know which opportunity to connect the competition to, we reference the CFDA record which links the two tables in the Oracle DB. We cannot use our assistance listing number table as we skip transforming opportunity assistance listings with null program title which only existed for the purposes of connecting the tables.
```
Feel free to adjust - just want future developers to know why we're doing this weird logic.